### ISSUE TITLE
Installation on Windows fails due to expired certificate of download.pytorch.org

### ISSUE THREAD

**Issue by ZoranPandovski** [pytorch/pytorch#33292]
## 🐛 Bug
Installation of torch and torchvision fails on windows.

## To Reproduce

1. Install on windows
2. RUN pip install torch===1.4.0 torchvision===0.5.0 -f https://download.pytorch.org/whl/torch_stable.html

Error log:
```
WARNING: Retrying (Retry(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by 'SSLError(SSLCertVerificationError(1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: certificate has expired (_ssl.c:1076)'))': /whl/torch_stable.html
Could not fetch URL https://download.pytorch.org/whl/torch_stable.html: There was a problem confirming the ssl certificate: HTTPSConnectionPool(host='download.pytorch.org', port=443): Max retries exceeded with url: /whl/torch_stable.html (Caused by SSLError(SSLCertVerificationError(1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: certificate has expired (_ssl.c:1076)'))) - skipping
ERROR: Could not find a version that satisfies the requirement torch===1.4.0 (from versions: 0.1.2, 0.1.2.post1, 0.1.2.post2)
ERROR: No matching distribution found for torch===1.4.0

```

cc @ezyang @gchanan @zou3519

**Comment by JACOBIN-SCTCS** [comment]
I too am having the same problem on Ubuntu

**Comment by xen0f0n** [comment]
![pytorch_ssl_2](https://user-images.githubusercontent.com/4368755/74453344-f751ad80-4e8a-11ea-8f31-2cb6b270187d.png)
Their SSL certificate expired today... You can download the .whl you need from [here](http://download.pytorch.org/whl/torch_stable.html) and install it using pip.

**Comment by xen0f0n** [comment]
@ZoranPandovski Also, you can replace https://download.pytorch.org/whl/torch_stable.html with https://s3.amazonaws.com/pytorch/whl/torch_stable.html and it will work.

**Comment by soumith** [comment]
The SSL Certificate issue is fixed now. The Cert was expired for ~4 hours, and we root-caused the auto-renewal problem, it wont happen again.

**Comment by joepareti54** [comment]
same problem as reported above, in my case on windows 10:
ERROR: Could not find a version that satisfies the requirement torch==1.4.0+cpu (from versions: none)
ERROR: No matching distribution found for torch==1.4.0+cpu
WARNING: pip is configured with locations that require TLS/SSL, however the ssl module in Python is not available.
Could not fetch URL https://pypi.org/simple/pip/: There was a problem confirming the ssl certificate: HTTPSConnectionPool(host='pypi.org', port=443): Max retries exceeded with url: /simple/pip/ (Caused by SSLError("Can't connect to HTTPS URL because the SSL module is not available.")) - skipping
Could not fetch URL https://download.pytorch.org/whl/torch_stable.html: There was a problem confirming the ssl certificate: HTTPSConnectionPool(host='download.pytorch.org', port=443): Max retries exceeded with url: /whl/torch_stable.html (Caused by SSLError("Can't connect to HTTPS URL because the SSL module is not available.")) - skipping

**Comment by svideloc** [comment]
Hi, I am having the same issue downloading torch=1.5.1+cpu for linux, any help on this?

ERROR: Could not find a version that satisfies the requirement torch==1.5.1+cpu (from versions: 0.1.2, 0.1.2.post1, 0.1.2.post2, 0.3.1, 0.4.0, 0.4.1, 1.0.0, 1.0.1, 1.0.1.post2, 1.1.0, 1.2.0, 1.3.0, 1.3.1, 1.4.0, 1.5.0, 1.5.1)
ERROR: No matching distribution found for torch==1.5.1+cpu

**Comment by ezyang** [comment]
@svideloc what command did you use to install, and where did you get the instruction from?

**Comment by svideloc** [comment]
Hey, I actually got it to work, I was having issues with download.pytorch.org not being a trusted host , so I added the below line to my pip config & it worked:

> trusted-host = download.pytorch.org

**Comment by emilianozm24** [comment]
> Hey, I actually got it to work, I was having issues with download.pytorch.org not being a trusted host , so I added the below line to my pip config & it worked:
> 
> > trusted-host = download.pytorch.org

How did you do this? I'm having this same issue with Windows 10 and nothing seems to work.

**Comment by svideloc** [comment]
if you are seeing the same issue as me, I was able to download when I made pytorch.org a trusted host in my pipconfig file, see the line of code below: 

> trusted-host = download.pytorch.org

**Comment by emilianozm24** [comment]
Sorry if this seems to amateur, but how did you edit your config file? Or did you just add it to the CLI like: " pip install ... trusted-host = download.pytorch.org ... "?

**Comment by svideloc** [comment]
You should be able to do that as well. If you want to change your pip config file there should be a directory .pip (it is hidden and it is in my home directory, I'm not sure where yours will be. Inside that dir, there is a pip.conf file, and you can just add the above line to it.

**Comment by Harryte** [comment]
This is the way to pass the --trusted-host in the pip install command:

pip install --trusted-host pypi.org --trusted-host pytorch.org --trusted-host download.pytorch.org --trusted-host files.pypi.org --trusted-host files.pytorch.org torch==1.8.1+cpu torchvision==0.9.1+cpu torchaudio===0.8.1 -f https://download.pytorch.org/whl/torch_stable.html

**Comment by eagertolearn1** [comment]
Following command works for me:

pip install --trusted-host=pypi.org --trusted-host=files.pythonhosted.org  --trusted-host=pytorch.org --trusted-host=download.pytorch.org --trusted-host=files.pypi.org --trusted-host=files.pytorch.org torch==1.8.1+cpu torchvision==0.9.1+cpu -f https://download.pytorch.org/whl/torch_stable.html

**Comment by ninofiliu** [comment]
Looks like this error is back

```console
(venv) PS D:\Hunyuan3D-2> pip3 install torch torchvision torchaudio --index-url https://download.pytorch.org/whl/cu126
Looking in indexes: https://download.pytorch.org/whl/cu126
WARNING: Retrying (Retry(total=4, connect=None, read=None, redirect=None, status=None)) after connection broken by 'SSLError(SSLCertVerificationError(1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: certificate has expired (_ssl.c:1007)'))': /whl/cu126/torch/
WARNING: Retrying (Retry(total=3, connect=None, read=None, redirect=None, status=None)) after connection broken by 'SSLError(SSLCertVerificationError(1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: certificate has expired (_ssl.c:1007)'))': /whl/cu126/torch/
WARNING: Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by 'SSLError(SSLCertVerificationError(1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: certificate has expired (_ssl.c:1007)'))': /whl/cu126/torch/
WARNING: Retrying (Retry(total=1, connect=None, read=None, redirect=None, status=None)) after connection broken by 'SSLError(SSLCertVerificationError(1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: certificate has expired (_ssl.c:1007)'))': /whl/cu126/torch/
WARNING: Retrying (Retry(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by 'SSLError(SSLCertVerificationError(1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: certificate has expired (_ssl.c:1007)'))': /whl/cu126/torch/
Could not fetch URL https://download.pytorch.org/whl/cu126/torch/: There was a problem confirming the ssl certificate: HTTPSConnectionPool(host='download.pytorch.org', port=443): Max retries exceeded with url: /whl/cu126/torch/ (Caused by SSLError(SSLCertVerificationError(1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: certificate has expired (_ssl.c:1007)'))) - skipping
ERROR: Could not find a version that satisfies the requirement torch (from versions: none)
ERROR: No matching distribution found for torch

[notice] A new release of pip is available: 23.0.1 -> 25.0.1
[notice] To update, run: python.exe -m pip install --upgrade pip
```

Fixed it with

```sh
 pip install torchaudio --index-url https://download.pytorch.org/whl/cu126 --trusted-host download.pytorch.org
```

because I'm assuming that the torch team is facing the same issue as four years ago https://github.com/pytorch/pytorch/issues/33292#issuecomment-585849399
